test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+                "UPAYLOAD_FORMAT_JSON" => UPayloadFormat::UPAYLOAD_FORMAT_JSON,
+                "UPAYLOAD_FORMAT_PROTOBUF" => UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF,
+                "UPAYLOAD_FORMAT_RAW" => UPayloadFormat::UPAYLOAD_FORMAT_RAW,
+                "UPAYLOAD_FORMAT_SOMEIP" => UPayloadFormat::UPAYLOAD_FORMAT_SOMEIP,
+
+                "UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY" => {
+                    UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY
+                }
+                "UPAYLOAD_FORMAT_SOMEIP_TLV" => UPayloadFormat::UPAYLOAD_FORMAT_SOMEIP_TLV,
+                "UPAYLOAD_FORMAT_TEXT" => UPayloadFormat::UPAYLOAD_FORMAT_TEXT,
+                "UPAYLOAD_FORMAT_UNSPECIFIED" => UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED,
+                _ => UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED,
+            }
+        }
+
+        let _format = match value.get("format") {
+            Some(_format) => from_str_format(
+                _format
+                    .as_str()
+                    .expect("Deserialize:something wrong with _type field"),
+            ),
+            None => UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED,
+        };
+
+        let _length = match value.get("length") {
+            Some(_length) => _length
+                .as_str()
+                .unwrap_or_else(|| panic!("Deserialize: something wrong with commstatus field"))
+                .parse::<i32>()
+                .expect("commstatus parsing error"),
+            None => 0,
+        };
+
+        
+         
+        let _data = match value.get("value") {
+            Some(_data) => Data::Value( serde_json::to_vec(_data).expect("error in converting data value to vector")),
+            None => Data::Reference(0),
+        };
+
+        let _special_fields = SpecialFields::default();
+
+        Ok(WrapperUPayload(UPayload {
+            length: Some(_length),
+            format: _format.into(),
+            data: _data.into(),
+            special_fields: _special_fields,
+        }))
+    }
+}
+
+#[derive(Debug, Default)]
+
+pub struct WrapperUMessage(pub UMessage);
+
+impl<'de> Deserialize<'de> for WrapperUMessage {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+
+        let wattributes = match value.get("attributes") {
+            Some(attributes) => {
+                serde_json::from_value::<WrapperUAttribute>(attributes.clone()).unwrap_or_default()
+            }
+            None => WrapperUAttribute(UAttributes::default()),
+        };
+
+        let wpayload = match value.get("payload") {
+            Some(payload) => {
+                serde_json::from_value::<WrapperUPayload>(payload.clone()).unwrap_or_default()
+            }
+            None => WrapperUPayload(UPayload::default()),
+        };
+
+        let mattribute = MessageField::from_option(Some(wattributes.0));
```

Comment:
You can avoid having to explicitly call `MessageField::from_option()` if down there when you're creating the `UMessage`, you use the `Into` trait:
```rust
Ok(WrapperUMessage(UMessage {
    attributes: Some(wattributes.0).into(),
    payload: Some(wpayload.0).into(),
    ..Default::default()
}
```

Reply:
corrected